microbench/src/main/java/io/netty/handler/codec/CodecOutputListBenchmark.java

```diff
@@ -54,16 +45,19 @@ public void destroy() {
 
     @Benchmark
     public void codecOutList() {
+        codecOutputList = CodecOutputList.newInstance();
         benchmarkAddAndClear(codecOutputList, elements);
```

Comment:
Shouldn't we call `codecOutputList.recycle()` here  and remove it from the `@TearDown` ?

Reply:
I thought about doing that on my first patch, but that would actually change what is being measured, and I couldn't do that without checking with you (the developers).

The way it was first implemented, the `destroy` method is called just **once** per entire benchmark execution (no Level on `@Teardown` specified, the default is Level.Trial). I think the call to the recycle does not seems to make sense here we are creating a new instance on every benchmark invocation.

Now if you also want to measure the `recycle` method call - which wasn't being measured before - I will include it there, otherwise I would actually remove it from the benchmark.

What do you think?

PS: The same analysis is true for the RecyclableArrayList

Reply:
true... lets just keep it as it is for now